pipeline/trusted-resources/pkg/trustedtask/sign_test.go

```diff
@@ -0,0 +1,261 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trustedtask
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"io"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"github.com/google/go-containerregistry/pkg/registry"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/test/diff"
+)
+
+func TestSignTaskSpec(t *testing.T) {
+	ctx := context.Background()
+	sv, err := GetSignerVerifier(password)
+	if err != nil {
+		t.Fatalf("failed to get signerverifier %v", err)
+	}
+
+	tcs := []struct {
+		name     string
+		signer   signature.SignerVerifier
+		taskSpec *v1beta1.TaskSpec
+		wantErr  bool
+	}{{
+		name:     "Sign TaskSpec",
+		signer:   sv,
+		taskSpec: taskSpecTest,
+		wantErr:  false,
+	}, {
+		name:     "Empty TaskSpec",
+		signer:   sv,
+		taskSpec: nil,
+		wantErr:  false,
+	}, {
+		name:     "Empty Signer",
+		signer:   nil,
+		taskSpec: taskSpecTest,
+		wantErr:  true,
+	},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			sig, err := SignTaskSpec(tc.signer, tc.taskSpec)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("SignTaskSpec() get err %v, wantErr %t", err, tc.wantErr)
+			}
+
+			if tc.wantErr {
+				return
+			}
+			signature, err := base64.StdEncoding.DecodeString(sig)
+			if err != nil {
+				t.Fatalf("error decoding signature: %v", err)
+			}
+			if err := VerifyTaskSpec(ctx, tc.taskSpec, tc.signer, signature); err != nil {
+				t.Fatalf("SignTaskSpec() generate wrong signature: %v", err)
+			}
+
+		})
+	}
+}
+
+func SignTaskSpec_MockSigner(t *testing.T) {
+	var signer MockSigner
+
+	tcs := []struct {
+		name     string
+		signer   signature.Signer
+		target   interface{}
+		expected string
+	}{{
+		name:     "Sign String",
+		signer:   signer,
+		target:   "Hello world",
+		expected: "tY805zV53PtwDarK3VD6dQPx5MbIgctNcg/oSle+MG0=",
+	}, {
+		name:     "Sign integer",
+		signer:   signer,
+		target:   12345,
```

Comment:
Because of the interface, there's actually not that much difference between the int vs string case - the test classes we would be more interested in are the ones covered above (e.g. nil signer, nil content, etc.). Perhaps we can merge these 2 tests?

Reply:
sure!